### PR TITLE
Add MachinePool resources

### PR DESCRIPTION
**Pull request by nprokopic**
### What does this PR do?

This PR adds machine pool resources and all other resources that are required to correctly deploy machine pools (more details here https://github.com/giantswarm/roadmap/issues/2742).

### What is the effect of this change to users?

Using this `cluster-$provider` apps will mean that very large portion of machine pools functionality is moved to a common chart.

Common codebase across all providers for (just the highlights):
- Operating system: starting to use this in onprem providers will mean that those providers switch from Ubuntu to Flatcar.
- Config of Kubernetes components: kubelet
- containerd config
- Commands that run before/after kubeadm on nodes
- Kubernetes version

### How does it look like?

Rendered by CI in one of the next PRs where the CI is fixed:

```
    ---
    # Source: cluster/templates/clusterapi/workers/kubeadmconfig.yaml
    apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
    kind: KubeadmConfig
    metadata:
      name: awesome-def00
      namespace: org-giantswarm
      annotations:
        machine-pool.giantswarm.io/name: awesome-def00
        important-cluster-value: 1000
        robots-need-this-in-the-cluster: eW91IGNhbm5vdCByZWFkIHRoaXMsIGJ1dCByb2JvdHMgY2FuCg==
        for-robots-in-nodepool: cm9ib3RzIGFyZSBvcGVyYXRpbmcgb24gdGhpcyBub2RlIHBvb2wK
      labels:
        app: cluster
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/version: 0.1.0-dev
        application.giantswarm.io/team: turtles
        giantswarm.io/cluster: awesome
        giantswarm.io/organization: giantswarm
        giantswarm.io/service-priority: highest
        cluster.x-k8s.io/cluster-name: awesome
        cluster.x-k8s.io/watch-filter: capi
        helm.sh/chart: cluster-0.1.0-dev
        another-cluster-label: label-2
        some-cluster-label: label-1
        giantswarm.io/machine-pool: awesome-def00
        nodepool-workload-type: ai
    spec:
      format: ignition
      ignition:
        containerLinuxConfig:
          additionalConfig: |
            systemd:
              units:      
              - name: kubeadm.service
                dropins:
                - name: 10-flatcar.conf
                  contents: |
                    [Unit]
                    # kubeadm must run after coreos-metadata populated /run/metadata directory.
                    Requires=coreos-metadata.service
                    After=coreos-metadata.service
                    [Service]
                    # Ensure kubeadm service has access to kubeadm binary in /opt/bin on Flatcar.
                    Environment=PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/opt/bin
                    # To make metadata environment variables available for pre-kubeadm commands.
                    EnvironmentFile=/run/metadata/*
              - name: example1.service
                enabled: false
                mask: false
                contents: |
                  # Contents goes here
                dropins:
                - name: hello.conf
                  contents: |
                    # Contents goes here
              - name: example2.service
                enabled: false
                mask: false
                contents: |
                  # Multi-line
                  # contents goes here
                dropins:
                - name: hello1.conf
                  contents: |
                    # Multi-line
                    # contents goes here
                - name: hello2.conf
                  contents: |
                    # Multi-line
                    # contents goes here
              - name: example1-workers.service
                enabled: false
                mask: false
                contents: |
                  # Contents goes here
                dropins:
                - name: hello.conf
                  contents: |
                    # Contents goes here
              - name: example2-workers.service
                enabled: false
                mask: false
                contents: |
                  # Multi-line
                  # contents goes here
                dropins:
                - name: hello1-workers.conf
                  contents: |
                    # Multi-line
                    # contents goes here
                - name: hello2-workers.conf
                  contents: |
                    # Multi-line
                    # contents goes here
            storage:
              directories:      
              - path: /var/lib/kubelet/temporary/stuff
                overwrite: true
                filesystem: kubelet
                mode: 750
                user:
                  id: 12345
                  name: giantswarm
                group:
                  id: 23456
                  name: giantswarm
              - path: /var/lib/kubelet
                mode: 750
              - path: /var/lib/kubelet/temporary/stuff/workers
                overwrite: true
                filesystem: kubelet
                mode: 750
                user:
                  id: 12345
                  name: giantswarm
                group:
                  id: 23456
                  name: giantswarm
            
      joinConfiguration:
        nodeRegistration:
          name: ${COREOS_EC2_HOSTNAME}
          kubeletExtraArgs:
            cloud-provider: external
            feature-gates: CronJobTimeZone=true
            healthz-bind-address: 0.0.0.0
            node-ip: ${COREOS_EC2_IPV4_LOCAL}
            node-labels: "role=worker,giantswarm.io/machine-pool=awesome-def00,map[workload-type:ai]"
            v: 2
          taints:
          - key: supernodepool
            value: hello
            effect: NoSchedule
      preKubeadmCommands:
      - "envsubst < /etc/kubeadm.yml > /etc/kubeadm.yml.tmp"
      - "mv /etc/kubeadm.yml.tmp /etc/kubeadm.yml"
      - "systemctl restart sshd"
      - "export HTTP_PROXY=http://proxy.giantswarm.io"
      - "export HTTPS_PROXY=https://proxy.giantswarm.io"
      - "export NO_PROXY="127.0.0.1,localhost,svc,local,awesome.example.gigantic.io,172.31.0.0/16,100.64.0.0/12,elb.amazonaws.com,169.254.169.254,some.noproxy.address.giantswarm.io,another.noproxy.address.giantswarm.io""
      - "export http_proxy=http://proxy.giantswarm.io"
      - "export https_proxy=https://proxy.giantswarm.io"
      - "export no_proxy="127.0.0.1,localhost,svc,local,awesome.example.gigantic.io,172.31.0.0/16,100.64.0.0/12,elb.amazonaws.com,169.254.169.254,some.noproxy.address.giantswarm.io,another.noproxy.address.giantswarm.io""
      - "echo "hello all nodes before kubeadm""
      - "echo "all cluster nodes before kubeadm""
      - "echo "hello workers before kubeadm""
      - "echo "cluster workers before kubeadm""
      postKubeadmCommands:
      - "/bin/sh /opt/kubelet-config.sh"
      - "echo "hello all nodes after kubeadm""
      - "echo "all cluster nodes after kubeadm""
      - "echo "hello workers after kubeadm""
      - "echo "cluster workers after kubeadm""
      users:
      - name: giantswarm
        groups: sudo
        sudo: "ALL=(ALL) NOPASSWD:ALL"
      files:
      - path: /etc/sysctl.d/hardening.conf
        permissions: 0644
        encoding: base64
        content: ZnMuaW5vdGlmeS5tYXhfdXNlcl93YXRjaGVzID0gMTYzODQKZnMuaW5vdGlmeS5tYXhfdXNlcl9pbnN0YW5jZXMgPSA4MTkyCmtlcm5lbC5rcHRyX3Jlc3RyaWN0ID0gMgprZXJuZWwuc3lzcnEgPSAwCm5ldC5pcHY0LmNvbmYuYWxsLmxvZ19tYXJ0aWFucyA9IDEKbmV0LmlwdjQuY29uZi5hbGwuc2VuZF9yZWRpcmVjdHMgPSAwCm5ldC5pcHY0LmNvbmYuZGVmYXVsdC5hY2NlcHRfcmVkaXJlY3RzID0gMApuZXQuaXB2NC5jb25mLmRlZmF1bHQubG9nX21hcnRpYW5zID0gMQpuZXQuaXB2NC50Y3BfdGltZXN0YW1wcyA9IDAKbmV0LmlwdjYuY29uZi5hbGwuYWNjZXB0X3JlZGlyZWN0cyA9IDAKbmV0LmlwdjYuY29uZi5kZWZhdWx0LmFjY2VwdF9yZWRpcmVjdHMgPSAwCiMgSW5jcmVhc2VkIG1tYXBmcyBiZWNhdXNlIHNvbWUgYXBwbGljYXRpb25zLCBsaWtlIEVTLCBuZWVkIGhpZ2hlciBsaW1pdCB0byBzdG9yZSBkYXRhIHByb3Blcmx5CnZtLm1heF9tYXBfY291bnQgPSAyNjIxNDQKIyBSZXNlcnZlZCB0byBhdm9pZCBjb25mbGljdHMgd2l0aCBrdWJlLWFwaXNlcnZlciwgd2hpY2ggYWxsb2NhdGVzIHdpdGhpbiB0aGlzIHJhbmdlCm5ldC5pcHY0LmlwX2xvY2FsX3Jlc2VydmVkX3BvcnRzPTMwMDAwLTMyNzY3Cm5ldC5pcHY0LmNvbmYuYWxsLnJwX2ZpbHRlciA9IDEKbmV0LmlwdjQuY29uZi5hbGwuYXJwX2lnbm9yZSA9IDEKbmV0LmlwdjQuY29uZi5hbGwuYXJwX2Fubm91bmNlID0gMgoKIyBUaGVzZSBhcmUgcmVxdWlyZWQgZm9yIHRoZSBrdWJlbGV0ICctLXByb3RlY3Qta2VybmVsLWRlZmF1bHRzJyBmbGFnCiMgU2VlIGh0dHBzOi8vZ2l0aHViLmNvbS9naWFudHN3YXJtL2dpYW50c3dhcm0vaXNzdWVzLzEzNTg3CnZtLm92ZXJjb21taXRfbWVtb3J5PTEKa2VybmVsLnBhbmljPTEwCmtlcm5lbC5wYW5pY19vbl9vb3BzPTEK
      - path: /etc/ssh/trusted-user-ca-keys.pem
        permissions: 0600
        encoding: base64
        content: c3NoLWVkMjU1MTkgQUFBQUMzTnphQzFsWkRJMU5URTVBQUFBSU00Y3ZaMDFmTG1POWNKYldVajdzZkYrTmhFQ2d5K0NsMGJhelNyWlg3c1UgdmF1bHQtY2FAdmF1bHQub3BlcmF0aW9ucy5naWFudHN3YXJtLmlvCg==
      - path: /etc/ssh/sshd_config
        permissions: 0600
        encoding: base64
        content: IyBVc2UgbW9zdCBkZWZhdWx0cyBmb3Igc3NoZCBjb25maWd1cmF0aW9uLgpTdWJzeXN0ZW0gc2Z0cCBpbnRlcm5hbC1zZnRwCkNsaWVudEFsaXZlSW50ZXJ2YWwgMTgwClVzZUROUyBubwpVc2VQQU0geWVzClByaW50TGFzdExvZyBubyAjIGhhbmRsZWQgYnkgUEFNClByaW50TW90ZCBubyAjIGhhbmRsZWQgYnkgUEFNCiMgTm9uIGRlZmF1bHRzICgjMTAwKQpDbGllbnRBbGl2ZUNvdW50TWF4IDIKUGFzc3dvcmRBdXRoZW50aWNhdGlvbiBubwpUcnVzdGVkVXNlckNBS2V5cyAvZXRjL3NzaC90cnVzdGVkLXVzZXItY2Eta2V5cy5wZW0KTWF4QXV0aFRyaWVzIDUKTG9naW5HcmFjZVRpbWUgNjAKQWxsb3dUY3BGb3J3YXJkaW5nIG5vCkFsbG93QWdlbnRGb3J3YXJkaW5nIG5vCkNBU2lnbmF0dXJlQWxnb3JpdGhtcyBlY2RzYS1zaGEyLW5pc3RwMjU2LGVjZHNhLXNoYTItbmlzdHAzODQsZWNkc2Etc2hhMi1uaXN0cDUyMSxzc2gtZWQyNTUxOSxyc2Etc2hhMi01MTIscnNhLXNoYTItMjU2LHNzaC1yc2EK
      - path: /etc/containerd/config.toml
        permissions: 0644
        contentFrom:
          secret:
            name: awesome-registry-configuration
            key: registry-config.toml
      - path: /etc/systemd/system/containerd.service.d/http-proxy.conf
        permissions: 0644
        encoding: base64
        content: W1NlcnZpY2VdCkVudmlyb25tZW50PSJIVFRQX1BST1hZPWh0dHA6Ly9wcm94eS5naWFudHN3YXJtLmlvIgpFbnZpcm9ubWVudD0iSFRUUFNfUFJPWFk9aHR0cHM6Ly9wcm94eS5naWFudHN3YXJtLmlvIgpFbnZpcm9ubWVudD0iTk9fUFJPWFk9MTI3LjAuMC4xLGxvY2FsaG9zdCxzdmMsbG9jYWwsYXdlc29tZS5leGFtcGxlLmdpZ2FudGljLmlvLDE3Mi4zMS4wLjAvMTYsMTAwLjY0LjAuMC8xMixlbGIuYW1hem9uYXdzLmNvbSwxNjkuMjU0LjE2OS4yNTQsc29tZS5ub3Byb3h5LmFkZHJlc3MuZ2lhbnRzd2FybS5pbyxhbm90aGVyLm5vcHJveHkuYWRkcmVzcy5naWFudHN3YXJtLmlvIgpFbnZpcm9ubWVudD0iaHR0cF9wcm94eT1odHRwOi8vcHJveHkuZ2lhbnRzd2FybS5pbyIKRW52aXJvbm1lbnQ9Imh0dHBzX3Byb3h5PWh0dHBzOi8vcHJveHkuZ2lhbnRzd2FybS5pbyIKRW52aXJvbm1lbnQ9Im5vX3Byb3h5PTEyNy4wLjAuMSxsb2NhbGhvc3Qsc3ZjLGxvY2FsLGF3ZXNvbWUuZXhhbXBsZS5naWdhbnRpYy5pbywxNzIuMzEuMC4wLzE2LDEwMC42NC4wLjAvMTIsZWxiLmFtYXpvbmF3cy5jb20sMTY5LjI1NC4xNjkuMjU0LHNvbWUubm9wcm94eS5hZGRyZXNzLmdpYW50c3dhcm0uaW8sYW5vdGhlci5ub3Byb3h5LmFkZHJlc3MuZ2lhbnRzd2FybS5pbyIK
      - path: /etc/systemd/system/kubelet.service.d/http-proxy.conf
        permissions: 0644
        encoding: base64
        content: W1NlcnZpY2VdCkVudmlyb25tZW50PSJIVFRQX1BST1hZPWh0dHA6Ly9wcm94eS5naWFudHN3YXJtLmlvIgpFbnZpcm9ubWVudD0iSFRUUFNfUFJPWFk9aHR0cHM6Ly9wcm94eS5naWFudHN3YXJtLmlvIgpFbnZpcm9ubWVudD0iTk9fUFJPWFk9MTI3LjAuMC4xLGxvY2FsaG9zdCxzdmMsbG9jYWwsYXdlc29tZS5leGFtcGxlLmdpZ2FudGljLmlvLDE3Mi4zMS4wLjAvMTYsMTAwLjY0LjAuMC8xMixlbGIuYW1hem9uYXdzLmNvbSwxNjkuMjU0LjE2OS4yNTQsc29tZS5ub3Byb3h5LmFkZHJlc3MuZ2lhbnRzd2FybS5pbyxhbm90aGVyLm5vcHJveHkuYWRkcmVzcy5naWFudHN3YXJtLmlvIgpFbnZpcm9ubWVudD0iaHR0cF9wcm94eT1odHRwOi8vcHJveHkuZ2lhbnRzd2FybS5pbyIKRW52aXJvbm1lbnQ9Imh0dHBzX3Byb3h5PWh0dHBzOi8vcHJveHkuZ2lhbnRzd2FybS5pbyIKRW52aXJvbm1lbnQ9Im5vX3Byb3h5PTEyNy4wLjAuMSxsb2NhbGhvc3Qsc3ZjLGxvY2FsLGF3ZXNvbWUuZXhhbXBsZS5naWdhbnRpYy5pbywxNzIuMzEuMC4wLzE2LDEwMC42NC4wLjAvMTIsZWxiLmFtYXpvbmF3cy5jb20sMTY5LjI1NC4xNjkuMjU0LHNvbWUubm9wcm94eS5hZGRyZXNzLmdpYW50c3dhcm0uaW8sYW5vdGhlci5ub3Byb3h5LmFkZHJlc3MuZ2lhbnRzd2FybS5pbyIK
    
    ---
    # Source: cluster/templates/clusterapi/workers/machinepool.yaml
    apiVersion: cluster.x-k8s.io/v1beta1
    kind: MachinePool
    metadata:
      annotations:
        helm.sh/resource-policy: keep
        machine-pool.giantswarm.io/name: awesome-def00
        important-cluster-value: 1000
        robots-need-this-in-the-cluster: eW91IGNhbm5vdCByZWFkIHRoaXMsIGJ1dCByb2JvdHMgY2FuCg==
        for-robots-in-nodepool: cm9ib3RzIGFyZSBvcGVyYXRpbmcgb24gdGhpcyBub2RlIHBvb2wK
      labels:
        app: cluster
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/version: 0.1.0-dev
        application.giantswarm.io/team: turtles
        giantswarm.io/cluster: awesome
        giantswarm.io/organization: giantswarm
        giantswarm.io/service-priority: highest
        cluster.x-k8s.io/cluster-name: awesome
        cluster.x-k8s.io/watch-filter: capi
        helm.sh/chart: cluster-0.1.0-dev
        another-cluster-label: label-2
        some-cluster-label: label-1
        giantswarm.io/machine-pool: awesome-def00
        nodepool-workload-type: ai
      name: awesome-def00
      namespace: org-giantswarm
    spec:
      clusterName: awesome
      replicas: 3
      template:
        spec:
          bootstrap:
            configRef:
              apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
              kind: KubeadmConfig
              name: awesome-def00
          clusterName: awesome
          infrastructureRef:
            apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
            kind: AWSMachinePool
            name: awesome-def00
          version: 1.24.10
```

### Any background context you can provide?

https://github.com/giantswarm/roadmap/issues/2742

### CODE REVIEW

File: helm/cluster/ci/ci-values.yaml [hunk 1]
```diff
@@ -59,64 +82,7 @@ internal:
...
-                    [Service]
-                    ExecStartPre=/bin/bash -c "while [ ! -f /etc/audit/rules.d/containerd.rules ]; do echo 'Waiting for /etc/audit/rules.d/containerd.rules to be written' && sleep 1; done"
-              - name: update-engine.service
-                enabled: false
-                mask: true
-              - name: locksmithd.service
-                enabled: false
-                mask: true
-              - name: example1.service
+              - name: example1-control-plane.service
```
Comment (nprokopic) [inline helm/cluster/ci/ci-values.yaml:85]: There were too many examples for the CI, removed most of them.

File: helm/cluster/templates/_helpers.tpl [hunk 2]
```diff
@@ -21,21 +21,37 @@ Given that Kubernetes allows 63 characters for resource names, the stem is trunc
...
+application.giantswarm.io/team: {{ index $.Chart.Annotations "application.giantswarm.io/team" | quote }}
 giantswarm.io/cluster: {{ include "cluster.resource.name" . | quote }}
 giantswarm.io/organization: {{ required "You must provide an existing organization name in .global.metadata.organization" $.Values.global.metadata.organization | quote }}
 giantswarm.io/service-priority: {{ $.Values.global.metadata.servicePriority }}
 cluster.x-k8s.io/cluster-name: {{ include "cluster.resource.name" . | quote }}
 cluster.x-k8s.io/watch-filter: capi
 helm.sh/chart: {{ include "cluster.chart.nameAndVersion" . | quote }}
 {{- end -}}
+
+{{- define "cluster.labels.custom" }}
```
Comment (nprokopic) [inline helm/cluster/templates/_helpers.tpl:43]: Named template for more easily adding common labels to all resources.

File: helm/cluster/templates/clusterapi/_helpers_files.tpl [hunk 3]
```diff
@@ -1,11 +1,10 @@
 {{- define "cluster.internal.kubeadm.files" }}
-{{- include "cluster.internal.kubeadm.files.sysctl" . }}
-{{- include "cluster.internal.kubeadm.files.systemd" . }}
-{{- include "cluster.internal.kubeadm.files.ssh" . }}
-{{- include "cluster.internal.kubeadm.files.cri" . }}
-{{- include "cluster.internal.kubeadm.files.kubelet" . }}
-{{- include "cluster.internal.kubeadm.files.kubernetes" . }}
-{{- include "cluster.internal.kubeadm.files.proxy" . }}
+{{- include "cluster.internal.kubeadm.files.sysctl" $ }}
```
Comment (nprokopic) [inline helm/cluster/templates/clusterapi/_helpers_files.tpl:2]: Minor change, passing the root context here. Currently it doesn't make a difference here, but doing it for the same of consistency with other `include` directives where it's needed.

File: helm/cluster/templates/clusterapi/_helpers_files.tpl [hunk 4]
```diff
@@ -61,19 +60,6 @@
 {{- end }}
 {{- end }}
 
-{{- define "cluster.internal.kubeadm.files.kubernetes" }}
```
Comment (nprokopic) [inline helm/cluster/templates/clusterapi/_helpers_files.tpl:64]: Moved this to control plane specific files, as it's not needed for worker nodes.

File: helm/cluster/templates/clusterapi/controlplane/_helpers_files.tpl [hunk 5]
```diff
@@ -1,9 +1,23 @@
...
+{{- include "cluster.internal.kubeadm.files.kubernetes" . }}
 {{- if $.Values.global.controlPlane.oidc.caPem }}
 - path: /etc/ssl/certs/oidc.pem
   permissions: "0600"
   encoding: base64
   content: {{ tpl ($.Files.Get "files/etc/ssl/certs/oidc.pem") . | b64enc }}
 {{- end }}
 {{- end }}
+
+{{- define "cluster.internal.kubeadm.files.kubernetes" }}
```
Comment (nprokopic) [inline helm/cluster/templates/clusterapi/controlplane/_helpers_files.tpl:12]: This template moved here to be control plane only, as it's not used on workers.

File: helm/cluster/templates/clusterapi/controlplane/_helpers_flatcar.tpl [hunk 6]
```diff
@@ -1,86 +1,39 @@
...
-  dropins:
-  {{- range .dropins }}
-  - name: {{ .name }}
-    {{- if .contents }}
-    contents: |
-      {{- .contents | trim | nindent 6 }}
-    {{- end }}
-  {{- end }}
-  {{- end }}
+{{- define "cluster.internal.controlPlane.kubeadm.ignition" }}
```
Comment (nprokopic) [inline helm/cluster/templates/clusterapi/controlplane/_helpers_flatcar.tpl:1]: Renamed template to align naming with other ones, and changed implementation to just pass the values to common Flatcar helpers that render systemd and storage config for Ignition.

File: helm/cluster/templates/clusterapi/controlplane/kubeadmcontrolplane.yaml [hunk 7]
```diff
@@ -22,16 +22,7 @@ spec:
...
-        additionalConfig: |
-          systemd:
-            units:
-            {{- include "cluster.kubeadmControlPlane.kubeadmConfigSpec.ignition.containerLinuxConfig.additionalConfig.systemd.units" $ | indent 12 }}
-          storage:
-            filesystems:
-            {{- include "cluster.kubeadmControlPlane.kubeadmConfigSpec.ignition.containerLinuxConfig.additionalConfig.storage.filesystems" $ | indent 12 }}
-            directories:
-            {{- include "cluster.kubeadmControlPlane.kubeadmConfigSpec.ignition.containerLinuxConfig.additionalConfig.storage.directories" $ | indent 12 }}
+      {{- include "cluster.internal.controlPlane.kubeadm.ignition" $ | indent 6 }}
```
Comment (nprokopic) [inline helm/cluster/templates/clusterapi/controlplane/kubeadmcontrolplane.yaml:25]: The template now includes the entire Ignition spec.

File: helm/cluster/templates/clusterapi/workers/_helpers_flatcar.tpl [hunk 8]
```diff
@@ -0,0 +1,37 @@
+{{- define "cluster.internal.workers.kubeadm.ignition" }}
```
Comment (nprokopic) [inline helm/cluster/templates/clusterapi/workers/_helpers_flatcar.tpl:1]: Worker-specific named templates for Flatcar systemd and storage config (these templates here just include generic templates and pass values for workers).

File: helm/cluster/templates/clusterapi/workers/kubeadmconfig.yaml [hunk 9]
```diff
@@ -0,0 +1,41 @@
+{{- if eq $.Values.internal.resourcesApi.nodePoolKind "MachinePool" }}
+{{- range $nodePoolName, $nodePoolConfig := $.Values.global.nodePools }}
+apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
```
Comment (nprokopic) [inline helm/cluster/templates/clusterapi/workers/kubeadmconfig.yaml:3]: KubeadmConfig for MachinePool.

File: helm/cluster/values.yaml [hunk 10]
```diff
@@ -43,11 +43,18 @@ internal:
           group: controlplane.cluster.x-k8s.io
           kind: KubeadmControlPlane
           version: v1beta1
+  kubeadmConfig:
```
Comment (nprokopic) [inline helm/cluster/values.yaml:46]: Some improvements in the JSON schema can make these default values not to be here (so it's less clutter on the eyes when looking at the values), but we can optimize these details later.

**Review by njuettner — APPROVED** [review]

**Pull request merged**